File: app/src/main/java/uk/lobsterdoodle/edinburghwolves/app/HomeActivity.java
package uk.lobsterdoodle.edinburghwolves.app;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import it.sephiroth.android.library.bottomnavigation.BottomNavigation;
import uk.lobsterdoodle.edinburghwolves.app.dummy.DummyContent;
import uk.lobsterdoodle.edinburghwolves.app.roster.RosterListFragment;
import uk.lobsterdoodle.edinburghwolves.app.roster.RosterListFragment.OnListFragmentInteractionListener;
import uk.lobsterdoodle.edinburghwolves.app.overview.OverviewFragment;

public class HomeActivity extends AppCompatActivity implements OverviewFragment.OnFragmentInteractionListener, OnListFragmentInteractionListener {

    private ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_wolves_home);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        viewPager = (ViewPager) findViewById(R.id.home_fragment_view_pager);
        viewPager.setAdapter(new HomeFragmentPagerAdapter(getSupportFragmentManager()));

        BottomNavigation bottomNavigation = (BottomNavigation) findViewById(R.id.nav_tabs);
        bottomNavigation.setOnMenuItemClickListener(new TabSelectedListener());
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }

    @Override
    public void onListFragmentInteraction(DummyContent.DummyItem item) {

    }

    private class TabSelectedListener implements BottomNavigation.OnMenuItemSelectionListener {

        @Override
        public void onMenuItemSelect(@IdRes int itemId, int position, boolean flag) {
            viewPager.setCurrentItem(position);
        }

        @Override
        public void onMenuItemReselect(@IdRes int itemId, int position, boolean flag) {
            viewPager.setCurrentItem(position);
        }
    }

    private class HomeFragmentPagerAdapter extends FragmentPagerAdapter {

        public HomeFragmentPagerAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public Fragment getItem(int position) {
            return position == 3
                        ? RosterListFragment.newInstance()
                        : OverviewFragment.newInstance();
        }
        @Override
        public int getCount() {
            return 5;
        }
    }
}

File: app/src/main/kotlin/uk/lobsterdoodle/edinburghwolves/home/HomeActivity.kt
package uk.lobsterdoodle.edinburghwolves.home

import android.net.Uri
import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import butterknife.BindView
import butterknife.ButterKnife
import uk.lobsterdoodle.edinburghwolves.OverviewFragment
import uk.lobsterdoodle.edinburghwolves.app.R
import uk.lobsterdoodle.edinburghwolves.app.overview.DummyFragment
import uk.lobsterdoodle.edinburghwolves.app.overview.DummyFragment.OnFragmentInteractionListener
import uk.lobsterdoodle.edinburghwolves.model.Fixture
import uk.lobsterdoodle.edinburghwolves.roster.RosterListFragment
import uk.lobsterdoodle.edinburghwolves.roster.RosterListFragment.OnListFragmentInteractionListener
import uk.lobsterdoodle.edinburghwolves.model.Player


class HomeActivity : AppCompatActivity(), OnFragmentInteractionListener, OnListFragmentInteractionListener, FixturesFragment.OnListFragmentInteractionListener {

    @BindView(R.id.home_fragment_view_pager) lateinit var viewPager: ViewPager
    @BindView(R.id.navigation) lateinit var navigation: BottomNavigationView

    private val listener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.nav_tab_item_home -> {
                viewPager.currentItem = 0
                return@OnNavigationItemSelectedListener true
            }
            R.id.nav_tab_item_fixtures -> {
                viewPager.currentItem = 1
                return@OnNavigationItemSelectedListener true
            }
            R.id.nav_tab_item_roster -> {
                viewPager.currentItem = 2
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_wolves_home)
        ButterKnife.bind(this)

        viewPager.adapter = HomeFragmentPagerAdapter(supportFragmentManager)
        navigation.setOnNavigationItemSelectedListener(listener)
    }

    private inner class HomeFragmentPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
        override fun getItem(position: Int): Fragment {
            when (position) {
                0 -> return OverviewFragment()
                1 -> return FixturesFragment()
                2 -> return RosterListFragment.newInstance()
                else -> { return DummyFragment.newInstance() }
            }
        }

        override fun getCount(): Int = 3
    }

    override fun onListFragmentInteraction(player: Player) {}

    override fun onListFragmentInteraction(fixture: Fixture) {}

    override fun onFragmentInteraction(uri: Uri?) {}
}
